File: cpp/TiketBioskop.cpp
#include "TiketBioskop.h"

// Constructor
TiketBioskop::TiketBioskop(
    string idTiket,
    string nomorKursi,
    int harga,
    string statusTiket,
    string judulFilm,
    string genre,
    int durasi,
    int ratingUsia,
    string studio,
    string jamTayang,
    string tanggalTayang,
    string jenisStudio
) : TiketFilm(
        idTiket,
        nomorKursi,
        harga,
        statusTiket,
        judulFilm,
        genre,
        durasi,
        ratingUsia
    ) {

    this->studio = studio;
    this->jamTayang = jamTayang;
    this->tanggalTayang = tanggalTayang;
    this->jenisStudio = jenisStudio;
}

// Getter
string TiketBioskop::getStudio() {
    return studio;
}

string TiketBioskop::getJamTayang() {
    return jamTayang;
}

string TiketBioskop::getTanggalTayang() {
    return tanggalTayang;
}

string TiketBioskop::getJenisStudio() {
    return jenisStudio;
}

// Setter
void TiketBioskop::setStudio(string studio) {
    this->studio = studio;
}

void TiketBioskop::setJamTayang(string jamTayang) {
    this->jamTayang = jamTayang;
}

void TiketBioskop::setTanggalTayang(string tanggalTayang) {
    this->tanggalTayang = tanggalTayang;
}

void TiketBioskop::setJenisStudio(string jenisStudio) {
    this->jenisStudio = jenisStudio;
}

File: php/TiketBioskop.php
<?php

require_once 'TiketFilm.php';

class TiketBioskop extends TiketFilm
{
    private $studio;
    private $jamTayang;
    private $tanggalTayang;
    private $jenisStudio;

    public function __construct(
        $idTiket,
        $nomorKursi,
        $harga,
        $statusTiket,
        $judulFilm,
        $genre,
        $durasi,
        $ratingUsia,
        $studio,
        $jamTayang,
        $tanggalTayang,
        $jenisStudio
    ) {
        parent::__construct(
            $idTiket,
            $nomorKursi,
            $harga,
            $statusTiket,
            $judulFilm,
            $genre,
            $durasi,
            $ratingUsia
        );

        $this->studio = $studio;
        $this->jamTayang = $jamTayang;
        $this->tanggalTayang = $tanggalTayang;
        $this->jenisStudio = $jenisStudio;
    }

    public function getStudio()
    {
        return $this->studio;
    }

    public function getJamTayang()
    {
        return $this->jamTayang;
    }

    public function getTanggalTayang()
    {
        return $this->tanggalTayang;
    }

    public function getJenisStudio()
    {
        return $this->jenisStudio;
    }

    public function setStudio($studio)
    {
        $this->studio = $studio;
    }

    public function setJamTayang($jamTayang)
    {
        $this->jamTayang = $jamTayang;
    }

    public function setTanggalTayang($tanggalTayang)
    {
        $this->tanggalTayang = $tanggalTayang;
    }

    public function setJenisStudio($jenisStudio)
    {
        $this->jenisStudio = $jenisStudio;
    }
}
?>